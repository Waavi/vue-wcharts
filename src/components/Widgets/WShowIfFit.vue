<template>
    <div
        v-if="!hide"
        class="WShowIfFit"
    >
        <slot />
    </div>
</template>

<script>
export default {
    name: 'WShowIfFit',
    data () {
        return {
            hide: false,
        }
    },
    mounted () {
        this.$nextTick(this.handleSize)
    },
    methods: {
        handleSize () {
            const { children = [], parentNode } = this.$el

            if (children.length) {
                const childrenWidth = Object.keys(children).reduce((a, b) => a + children[b].offsetWidth, 0)
                const parentWidth = parentNode.clientWidth
                this.hide = parentWidth < childrenWidth
            }
        },
    },
}
</script>
