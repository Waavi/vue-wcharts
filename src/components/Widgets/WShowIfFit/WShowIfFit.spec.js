import { shallowMount } from '@vue/test-utils'
import WShowIfFit from './WShowIfFit.vue'

describe('Widgets/WShowIfFit', () => {
    const propsData = {}
    const defaultConfig = {
        propsData,
        slots: {
            default: ['<div />'],
        },
    }

    it(`Should be render correctly`, () => {
        const wrapper = shallowMount(WShowIfFit, defaultConfig)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it(`Should be check if show or hide child component`, () => {
        const wrapper = shallowMount(WShowIfFit, defaultConfig)

        Object.defineProperty(wrapper.vm.$el, 'parentNode', {
            get () {
                return {
                    appendChild: () => undefined,
                    removeChild: () => undefined,
                    clientWidth: 25,
                }
            },
        })

        Object.defineProperty(wrapper.vm.$el, 'children', {
            get () {
                return [
                    {
                        appendChild: () => undefined,
                        removeChild: () => undefined,
                        offsetWidth: 50,
                    },
                ]
            },
        })

        wrapper.vm.handleSize()
        expect(wrapper.vm.hide).toBeTruthy()
        expect(wrapper.html()).toMatchSnapshot()
    })
})
