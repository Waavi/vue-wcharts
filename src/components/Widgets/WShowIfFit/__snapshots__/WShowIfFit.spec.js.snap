// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/WShowIfFit Should be check if show or hide child component 1`] = `undefined`;

exports[`Widgets/WShowIfFit Should be render correctly 1`] = `
<div class="WShowIfFit">
  <div></div>
</div>
`;
